Introduce new CSV Task

adrianna-chang-shopify opened the pull request:
Closes: https://github.com/Shopify/maintenance_tasks/issues/201

This PR introduces a new `CsvTask` module that Tasks can include in order to process a CSV instead of supplying a collection. CsvTask implements `collection` and `count`, and expects to be passed `csv_content` in the form of a `String`. `collection` is implemented by generating a new `CSV` with headers from `csv_content`, and `count` simply counts the number of new lines in the CSV.

etiennebarrie reviewed (commented):
I feel like we're missing the part tying together TaskJob and CsvTask, and also you've noted that we know show the code of CsvTask (since it holds the `collection` method) instead of the task itself, I think we should fix it while we're here.

adrianna-chang-shopify commented on test/dummy/app/tasks/maintenance/test_csv_task.rb:
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+module Maintenance
+  class TestCsvTask < MaintenanceTasks::Task
```
Yes, great idea!

adrianna-chang-shopify commented:
Okay @etiennebarrie , I've pulled in the changes tying together `TaskJob` with creating new `Tasks` that process CSVs.
I also changed the test CSV Task to match what we've documented in the readme, and updated `sample.csv` to reflect the changes. LMK what you think!

etiennebarrie reviewed (approved):
I ran
```
MaintenanceTasks::Runner.run name: "Maintenance::ImportPostsTask", csv_file: { io: File.open('test/fixtures/files/sample.csv'), filename: 'sample.csv' }
```
in a console and it worked!

rafaelfranca commented on app/tasks/maintenance_tasks/csv_task.rb:
```diff
@@ -0,0 +1,30 @@
+
+module MaintenanceTasks
+  # A mixin that provides a Task with the ability to process CSV files.
+  module CsvTask
```
It seems to me that this module doesn't need to exist and we can default `collection` and `count` to this behavior if `csv_content` is present in the task. The content is only present if the user upload to the run right?

Why make users have to add this module if the system can do everything automatically?
